kobo/apps/openrosa/apps/main/tests/test_past_bugs.py

```diff
@@ -14,20 +17,29 @@ class TestInputs(TestBase):
     def test_uniqueness_of_group_names_enforced(self):
         pre_count = XForm.objects.count()
         self._create_user_and_login()
-        response = self._publish_xls_file(
-            'fixtures/group_names_must_be_unique.xls')
-        message = 'The name "group_names_must_be_unique" is the same as the form name'
-        self.assertTrue(message in response.json()['text'])
-        self.assertEqual(XForm.objects.count(), pre_count)
 
-    def test_mch(self):
-        self._publish_xls_file('fixtures/bug_fixes/MCH_v1.xls')
+        with pytest.raises(PyXFormError) as e:
+            self._publish_xls_file(
+                'fixtures/group_names_must_be_unique.xls'
+            )
+            assert 'The name "group_names_must_be_unique" is the same as the form name' in str(e)
+
+        assert XForm.objects.count() == pre_count
+
+    def test_buggy_files(self):
+        message = "Unknown question type 'Select one from source'"
+        with pytest.raises(PyXFormError) as e:
+            self._publish_xls_file(
+                os.path.join('fixtures/bug_fixes/MCH_v1.xls')
+            )
+            assert message == str(e)
 
-    def test_erics_files(self):
-        for name in ['battery_life.xls',
-                     'enumerator_weekly.xls',
-                     'Enumerator_Training_Practice_Survey.xls']:
-            self._publish_xls_file(os.path.join('fixtures', 'bug_fixes', name))
+        message = 'Duplicate column header: label'
+        with pytest.raises(PyXFormError) as e:
+            self._publish_xls_file(
+                os.path.join('fixtures', 'bug_fixes', 'enumerator_weekly.xls')
+            )
+            assert message == str(e)
```

Comment:
for `MCH_v1.xls` you are right! But I think with previous tests, its deployment in tests was not hitting same parts of the code that production would. Thus, the error was silently ignored. 
If you try to deploy this project on production, it does return the same error.

Reply:
Eric's project: `enumerator_weekly.xls` is also buggy on production. 

I brought back `test_erics_files` test.

Reply:
We looked into it and `MCH_v1.xls` is a weird one. It uses an archaic `Select one from source` syntax, where `source` is a list name. Pyxform tolerates `select one from…`, but it's actually *case sensitive* (who knew?) and balks at `Select one from…`. We changed the XLSForm file to use lowercase `select one from source`.

It's unclear if that was the particular "bug" that was being tested initially? But we'll keep `MCH_v1.xls` around in case it saves us from some regression that's actually significant :shrug: